include/crm/common/strings_internal.h

```diff
@@ -16,6 +16,13 @@
 
 /* internal generic string functions (from strings.c) */
 
+enum pcmk__str_flags {
+    pcmk__str_none          = 0,
```

Comment:
I'm wondering whether we should name this pcmk__str_case (or cases) for readability, e.g. looking at

    pcmk__strcmp(s1, s2, pcmk__str_none)
    pcmk__strcmp(s1, s2, pcmk__str_case)
    pcmk__strcmp(s1, s2, pcmk__str_casei)

It would make every comparison explicitly case-sensitive or not. OTOH pcmk__str_case|pcmk__str_casei would be legal but odd, and the other flags aren't referenced in any way. I'm fine with it either way.